### PR TITLE
Support get/set as property name in object literal

### PR DESCRIPTION
**Pull request by rerobika**
JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/parser/js/js-parser-expr.c [hunk 1]
```diff
@@ -791,11 +791,21 @@ parser_parse_object_literal (parser_context_t *context_p) /**< context */
         uint32_t status_flags;
         cbc_ext_opcode_t opcode;
         uint16_t literal_index, function_literal_index;
+        bool is_getter = context_p->token.type == LEXER_PROPERTY_GETTER;
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser-expr.c]: There is no need to save this value, `lexer_check_next_character` does not change it.

File: jerry-core/parser/js/js-parser-expr.c [hunk 2]
```diff
@@ -791,11 +791,21 @@ parser_parse_object_literal (parser_context_t *context_p) /**< context */
...
 #if !ENABLED (JERRY_ES2015)
         parser_object_literal_item_types_t item_type;
+#else /* ENABLED (JERRY_ES2015) */
+        if (lexer_check_next_character (context_p, LIT_CHAR_LEFT_PAREN))
+        {
+          lexer_construct_literal_object (context_p,
+                                          (is_getter ? (lexer_lit_location_t *) &lexer_get_literal
+                                                     : (lexer_lit_location_t *) &lexer_set_literal),
+                                          LEXER_STRING_LITERAL);
+          goto parse_object_method;
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser-expr.c]: Can we avoid this goto?

Comment (rerobika) [reply jerry-core/parser/js/js-parser-expr.c]: Sure! I've reordered the switch cases to substitute the goto with a fallthru.

**Comment by zherczeg** [comment]
Just a question: would it be better to add `(` here:
https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/parser/js/js-lexer.c#L2474

**Pull request merged**